Doc/library/sys.rst

```diff
@@ -338,16 +338,16 @@ always available.
    +-----------------------------+----------------------------------------------+
    | Attribute                   | Explanation                                  |
    +=============================+==============================================+
-   | :const:`emscripten_version` | Emscripten version as tuple of ints          |
+   | :attr:`!emscripten_version` | Emscripten version as tuple of ints          |
    |                             | (major, minor, micro), e.g. ``(3, 1, 8)``.   |
    +-----------------------------+----------------------------------------------+
-   | :const:`runtime`            | Runtime string, e.g. browser user agent,     |
+   | :attr:`!runtime`            | Runtime string, e.g. browser user agent,     |
    |                             | ``'Node.js v14.18.2'``, or ``'UNKNOWN'``.    |
    +-----------------------------+----------------------------------------------+
-   | :const:`pthreads`           | ``True`` if Python is compiled with          |
+   | :attr:`!pthreads`           | ``True`` if Python is compiled with          |
    |                             | Emscripten pthreads support.                 |
    +-----------------------------+----------------------------------------------+
-   | :const:`shared_memory`      | ``True`` if Python is compiled with shared   |
+   | :attr:`!shared_memory`      | ``True`` if Python is compiled with shared   |
    |                             | memory support.                              |
    +-----------------------------+----------------------------------------------+
```

Comment:
I wonder if we could do something like this instead:

```rst
.. data:: _emscripten_info

   A :term:`named tuple` holding information about the environment on the
   *wasm32-emscripten* platform. The named tuple is provisional and may change
   in the future.

   .. attribute:: _emscripten_info.emscripten_version

      Emscripten version as tuple of ints (major, minor, micro), e.g. ``(3, 1, 8)``.

   .. attribute:: _emscripten_info.runtime

      Runtime string, e.g. browser user agent, ``'Node.js v14.18.2'``, or ``'UNKNOWN'``.

   .. attribute:: _emscripten_info.pthreads

      ``True`` if Python is compiled with Emscripten pthreads support.

   .. attribute:: _emscripten_info.shared_memory

      ``True`` if Python is compiled with shared memory support.

   .. availability:: Emscripten.

   .. versionadded:: 3.11
```

I'm not sure if this is how you're "supposed" to use the `.. attribute::` directive (since `_emscripten_info` is module-level data rather than a class). But it seems to render okay for me locally, and there are no warnings while Sphinx is building the docs:

![image](https://github.com/python/cpython/assets/66076021/478a5912-991a-4359-ae06-5083a9fa9838)

Reply:
(The advantage being that Sphinx adds an anchor for each separate attribute, which could then be linked to from other pieces of documentation)

Reply:
Could possibly do the same for `int_info`, `hash_info` and `thread_info` as well (but I'd wait to see if @AA-Turner thinks this general approach is okay before doing so!)

Reply:
> (The advantage being that Sphinx adds an anchor for each separate attribute, which could then be linked to from other pieces of documentation)

That's a nice side effect. We should do that for the other documented named tuples as well. See also my PR for `float_info`.

Reply:
Seems eminently reasonable; the anchors are formatted correctly, too (``#sys._emscripten_info.emscripten_version``, etc)

Reply:
```suggestion
   .. data:: _emscripten_info

      A :term:`named tuple` holding information about the environment on the
      *wasm32-emscripten* platform.
      The named tuple is provisional and may change in the future.

      .. attribute:: _emscripten_info.emscripten_version

         Emscripten version as tuple of ints (major, minor, micro), e.g. ``(3, 1, 8)``.

      .. attribute:: _emscripten_info.runtime

         Runtime string, e.g. browser user agent, ``'Node.js v14.18.2'``, or ``'UNKNOWN'``.

      .. attribute:: _emscripten_info.pthreads

         ``True`` if Python is compiled with Emscripten pthreads support.

      .. attribute:: _emscripten_info.shared_memory

         ``True`` if Python is compiled with shared memory support.

      .. availability:: Emscripten.

      .. versionadded:: 3.11
```